src/vm_stack.zig

```diff
@@ -2100,7 +2102,11 @@ const InstructionFuncs = struct {
         };
 
         if (num_pages >= 0 and memory_instance.grow(@as(usize, @intCast(num_pages)))) {
-            stack.pushI32(old_num_pages);
+            switch (memory_instance.limits.indexType()) {
+                .I32 => stack.pushI32(old_num_pages),
+                .I64 => stack.pushI64(old_num_pages),
+                else => unreachable,
+            }
```

Comment:
This was the culprit. Old size was being pushed as an i32, but read in storeInMem as an i64. Put the offset way out of the range of the memory.

Reply:
Ah good catch! I'll take a look at all the changes tonight and merge it soon. Thanks again for all the work.